From c47410e82446cf06633ffe6575b2e09a9595c985 Mon Sep 17 00:00:00 2001
From: Celica Sylphil <kamikitakomaric@protonmail.com>
Date: Sat, 7 Mar 2026 13:40:34 +0800
Subject: [PATCH] ANDROID: sched: Move SYSVIPC fields to the end of task_struct

---
 include/linux/sched.h | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/include/linux/sched.h b/include/linux/sched.h
index b275cd285182..a00ac008c101 100644
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -1073,10 +1073,6 @@ struct task_struct {
 
 	struct nameidata		*nameidata;
 
-#ifdef CONFIG_SYSVIPC
-	struct sysv_sem			sysvsem;
-	struct sysv_shm			sysvshm;
-#endif
 #ifdef CONFIG_DETECT_HUNG_TASK
 	unsigned long			last_switch_count;
 	unsigned long			last_switch_time;
@@ -1532,6 +1528,11 @@ struct task_struct {
 	 *
 	 * Do not put anything below here!
 	 */
+
+#ifdef CONFIG_SYSVIPC
+	struct sysv_sem			sysvsem;
+	struct sysv_shm			sysvshm;
+#endif
 };
 
 static inline struct pid *task_pid(struct task_struct *task)
-- 
2.34.1
